tests/summ_ai_api/summ_ai_test.py

```diff
@@ -224,3 +477,147 @@ async def test_summ_ai_error_handling(
+        in errors
+    ), "after check_rate_limit_exceeded() with 429 or 529, a logging entry should appear"
+
+    assert (
+        SummAiApiClient(MockedRequest(), "form_dummy").check_rate_limit_exceeded(500)
+        is False
+    ), "if response-status not in (429, 529), check-rate-limit should return False"
+
+
+def test_patient_task_queue_normal_deque():
+    """
+    Test for PatientTaskQueue Class. Tests that a first-in/first out (fifo) deque-queue is build from a task-list
+    Checks that next() removes one element each from the queue
+    """
+
+    tasks = [1, 2, 3]
+    task_generator = PatientTaskQueue(tasks, 1)
+
+    # we check that list comes reverse (fifo-queue)
+    for i in range(len(task_generator)):
+        assert next(task_generator) == tasks[i]
+
+    assert len(task_generator) == 0
+    # we check that next() did remove elements from the queue
+    with pytest.raises(StopIteration):
+        next(task_generator)
+
+
+async def test_patient_task_queue_hit_rate_limit():
+    """
+    Test for PatientTaskQueue Class. It tests that when the maximum number of parallel requests
+    to the summ_ai_api has been exceeded (rate limit exceeded), a defined time is waited until the next task object is returned via next().
+    """
+
+    # pylint: disable=missing-return-doc, missing-return-type-doc
+    async def identity(x):
+        return x
+
+    tasks = [
+        partial(identity, 1),
+        partial(identity, 2),
+        partial(identity, 3),
+    ]
+    # for the test, waiting-time is reduced to 0.5s, default is defined in settings.SUMM_AI_RATE_LIMIT_COOLDOWN
+    task_generator = PatientTaskQueue(tasks, 0.5)
+
+    # Rate limit hit
+
+    task = next(task_generator)
+    assert task in tasks, "PatientTaskQueue should return sane element"
+    # we call out a reach the rate limit
+    task_generator.hit_rate_limit(task)
+    assert (
+        task_generator.last_rate_limit is not None
+    ), "if hit_rate_limit() was called, last_rate_limit must not be None"
+    assert len(task_generator) == len(tasks)
+
+    # Next task is wait
+    task = next(task_generator)()
+    assert asyncio.iscoroutine(task)
+    assert (
+        task.__name__ == "sleep"
+    ), "while in wait, next() is expected to return a 'sleep'-task, with _name_==sleep."
+    # Cleanup coroutine without awaiting
+    task.close()
+
+    # wait 0.3s
+    await asyncio.sleep(0.3)
+
+    # Next task is still wait
+    task = next(task_generator)()
+    assert asyncio.iscoroutine(task)
+    assert (
+        task.__name__ == "sleep"
+    ), "while in wait, next() is expected to return a 'sleep'-task, with _name_==sleep."
+    # Cleanup coroutine without awaiting
+    task.close()
+
+    await asyncio.sleep(0.3)
+
+    # Next task is normal again
+    task = next(task_generator)
+    assert (
+        task in tasks
+    ), "when wait-mode passed, next() is expected to return the next task"
```

Comment:
```suggestion
    # Test first task
    task = await task_generator.__anext__()
    assert task == tasks[0], "PatientTaskQueue should return the first element"

    # Test second task with timing
    start = time.time()
    task = await task_generator.__anext__()
    end = time.time()
    assert task == tasks[1], "PatientTaskQueue should return the second element"
    assert (
        end - start < 0.5
    ), "PatientTaskQueue should return the second element very fast"

    # Hit the rate limit and check whether the second task is rescheduled
    task_generator.hit_rate_limit(task)
    start = time.time()
    task = await task_generator.__anext__()
    end = time.time()
    assert task == tasks[1], "PatientTaskQueue should return the second element again"
    assert end - start > 0.5, "PatientTaskQueue should return the second element slower"

    # Test third task
    start = time.time()
    task = await task_generator.__anext__()
    end = time.time()
    assert task == tasks[2], "PatientTaskQueue should return the third element"
    assert (
        end - start < 0.5
    ), "PatientTaskQueue should return the third element very fast"

    assert (
        len(task_generator) == 0
    ), "PatientTaskQueue should be empty after three tasks have been removed"
```
Here as well, we should test the async queue instead of the sync one.